code/asteroid/asteroid.cpp

```diff
@@ -1137,6 +1138,23 @@ int asteroid_check_collision(object *pasteroid, object *other_obj, vec3d *hitpos
 
 		}
 	}
+
+	// check if the hit point is beyond the clip plane if the ship is warping.
+	if (mc_ret_val) {
+		WarpEffect* warp_effect = nullptr;
+		ship* shipp = &Ships[pship_obj->instance];
+
+		// this is extremely confusing but mc.hit_point_world isn't actually in world coords
+		// everything above was calculated relative to the heavy's position
+		vec3d actual_world_hit_pos = mc.hit_point_world + heavy_obj->pos;
+		if ((shipp->is_arriving()) && (shipp->warpin_effect))
```

Comment:
Checking against nullptr or zero or both? Even if it's done similarly around this part of the code, probably better to explicitly check, if for no other reason than for readability.  You've probably picked up by now that that's one of my big things.

Also the compiler will probably optimize it for us.